vortex-array/src/array/struct_/compute.rs

```diff
@@ -71,3 +76,25 @@ impl SliceFn for StructArray {
         .map(|a| a.into_array())
     }
 }
+
+impl FilterFn for StructArray {
+    fn filter(&self, predicate: &Array) -> VortexResult<Array> {
+        let fields = self
+            .children()
+            .map(|field| filter(&field, predicate))
+            .try_collect()?;
+
+        let predicate_true_count = predicate
```

Comment:
Is it worth doing this only if you have no fields? `fields.first().map(|a| a.len()).or_else(...)`

Reply:
yes, I noticed this is what Arrow does as well

Reply:
I thought I pushed that, I'll add that to another PR